### PR TITLE
[Merged by Bors] - Inform peers of requests that exceed the maximum rate limit + log downgrade

### PR DESCRIPTION
**Pull request by divagant-martian**
## Issue Addressed

#1825 

## Proposed Changes

Since we penalize more blocks by range requests that have large steps, it is possible to get requests that will never be processed. We were not informing peers about this requests and also logging CRIT that is no longer relevant. Later we should check if more sophisticated handling for those requests is needed

### CODE REVIEW

File: beacon_node/eth2_libp2p/src/rpc/mod.rs [hunk 1]
```diff
@@ -233,9 +233,23 @@ where
                         }))
                 }
                 Err(RateLimitedErr::TooLarge) => {
-                    // we set the batch sizes, so this is a coding/config err
-                    crit!(self.log, "Batch too large to ever be processed";
-                        "protocol" => format!("{}", req.protocol()));
+                    // we set the batch sizes, so this is a coding/config err for most protocols
```
Comment (AgeManning) [inline beacon_node/eth2_libp2p/src/rpc/mod.rs:236]:
```suggestion
```

File: beacon_node/eth2_libp2p/src/rpc/mod.rs [hunk 2]
```diff
@@ -233,9 +233,23 @@ where
...
+                    let protocol = req.protocol();
+                    if matches!(protocol, Protocol::BlocksByRange) {
+                        debug!(self.log, "Blocks by range request will never be processed"; "request" => %req);
+                    } else {
+                        crit!(self.log, "Request size too large to ever be processed"; "protocol" => %protocol);
```
Comment (AgeManning) [inline beacon_node/eth2_libp2p/src/rpc/mod.rs:241]: I think we don't need this case right? I think its not just a coding error. 

We could hit this if for example someone asked a request with a maximum BlocksByRoot size right? i.e they fill the request with as many roots as the RPC allows, in principle we could hit this then right?

**Review by AgeManning — COMMENTED** [review]
Nice changes. 

Minor comments

File: beacon_node/eth2_libp2p/src/rpc/mod.rs (hunk 2 above)
Comment (divagant-martian) [reply beacon_node/eth2_libp2p/src/rpc/mod.rs:241]: the limit for BlocksByRoot is MAX_REQUEST_BLOCKS, and this error is triggered if the weight of the request is bigger than that. This error implies that we are either not allowing requests up to the largest allowed size or that the rpc is letting pass requests that do not comply. So I think this is indeed a CRIT

**Review by AgeManning — APPROVED** [review]
LGTM

**Pull request closed without merging**